workspace_host/interface.js

```diff
@@ -550,7 +540,7 @@ async function _getWorkspaceSettingsAsync(workspace_id) {
 }
 
 function _pullImage(workspace, callback) {
-  workspaceHelper.updateMessage(workspace.id, 'Checking image');
+  workspaceUtils.updateWorkspaceMessage(workspace.id, 'Checking image');
```

Comment:
I think it was probably an error to not be awaiting these calls. In practice it likely works anyway and the only real harm could be out-of-order message updates. For correctness I think it's better to await them, however.

There are a lot more of these in this file but I only marked this one to reduce commenting noise.

```suggestion
  await workspaceUtils.updateWorkspaceMessage(workspace.id, 'Checking image');
```

Reply:
Good catch. Hopefully it's not too bad to convert everything upstream of here to `async`!

Reply:
Done in 6a3f608 and 3ab471e. I touched some hairy callback code, so I'd appreciate a close look at that!

Reply:
These look good to me (read the code, did not test).